acvm_backend_barretenberg/src/composer.rs

```diff
@@ -167,8 +198,235 @@ impl StandardComposer {
+        use super::POINTER_BYTES;
+        use wasmer::Value;
+
+        let cs_buf = constraint_system.to_bytes();
+        let cs_ptr = self.allocate(&cs_buf);
+
+        // The proving key is not actually written to this pointer.
+        // `pk_ptr_ptr` is a pointer to a pointer which holds the proving key.
+        let pk_ptr_ptr: usize = 0;
+
+        let pk_size = self
+            .call_multiple(
+                "acir_proofs_init_proving_key",
+                vec![&cs_ptr, &Value::I32(pk_ptr_ptr as i32)],
+            )
+            .value();
+        let pk_size: usize = pk_size.unwrap_i32() as usize;
+
+        // We then need to read the pointer at `pk_ptr_ptr` to get the key's location
+        // and then slice memory again at `pk_ptr` to get the proving key.
+        let pk_ptr = self.slice_memory(pk_ptr_ptr, POINTER_BYTES);
+        let pk_ptr: usize =
+            u32::from_le_bytes(pk_ptr[0..POINTER_BYTES].try_into().unwrap()) as usize;
+
+        self.slice_memory(pk_ptr, pk_size)
+    }
+
+    fn compute_verification_key(
+        &self,
+        constraint_system: &ConstraintSystem,
+        proving_key: &[u8],
+    ) -> Vec<u8> {
+        use super::POINTER_BYTES;
+        use wasmer::Value;
+
+        let circuit_size = self.get_circuit_size(constraint_system);
+        let CRS {
+            g1_data, g2_data, ..
+        } = CRS::new(circuit_size as usize);
+        let pippenger_ptr = self.get_pippenger(&g1_data).pointer();
+
+        let g2_ptr = self.allocate(&g2_data);
+        let pk_ptr = self.allocate(proving_key);
+
+        // The verification key is not actually written to this pointer.
+        // `vk_ptr_ptr` is a pointer to a pointer which holds the verification key.
+        let vk_ptr_ptr: usize = 0;
+
+        let vk_size = self
+            .call_multiple(
+                "acir_proofs_init_verification_key",
+                vec![
+                    &pippenger_ptr,
+                    &g2_ptr,
+                    &pk_ptr,
+                    &Value::I32(vk_ptr_ptr as i32),
+                ],
+            )
+            .value();
+        let vk_size: usize = vk_size.unwrap_i32() as usize;
+
+        // We then need to read the pointer at `vk_ptr_ptr` to get the key's location
+        // and then slice memory again at `vk_ptr` to get the verification key.
+        let vk_ptr = self.slice_memory(vk_ptr_ptr, POINTER_BYTES);
+        let vk_ptr: usize =
+            u32::from_le_bytes(vk_ptr[0..POINTER_BYTES].try_into().unwrap()) as usize;
+
+        self.slice_memory(vk_ptr, vk_size)
+    }
+
+    fn create_proof_with_pk(
+        &self,
+        constraint_system: &ConstraintSystem,
+        witness: WitnessAssignments,
+        proving_key: &[u8],
+    ) -> Vec<u8> {
+        use super::POINTER_BYTES;
+        use wasmer::Value;
+
+        let circuit_size = self.get_circuit_size(constraint_system);
+        let CRS {
+            g1_data, g2_data, ..
+        } = CRS::new(circuit_size as usize);
```

Comment:
Does this actually use g1_data? Because it might be more efficient to use a G2::new instead